File: db4oj.tests/src/com/db4o/db4ounit/common/soda/classes/typedhierarchy/STRTH2.java
/* Copyright (C) 2004   Versant Inc.   http://www.db4o.com */

package com.db4o.db4ounit.common.soda.classes.typedhierarchy;

public class STRTH2{
	
	public STRTH1TestCase parent;
	public STRTH3 h3;
	public String foo2;
	
	public STRTH2(){
	}
	
	public STRTH2(STRTH3 a3){
		h3 = a3;
		a3.parent = this;
	}
	
	public STRTH2(String str){
		foo2 = str;
	}
	
	public STRTH2(STRTH3 a3, String str){
		h3 = a3;
		a3.parent = this;
		foo2 = str;
	}
}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Soda/Classes/Typedhierarchy/STRTH2.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Tests.Common.Soda.Classes.Typedhierarchy;

namespace Db4objects.Db4o.Tests.Common.Soda.Classes.Typedhierarchy
{
	public class STRTH2
	{
		public STRTH1TestCase parent;

		public STRTH3 h3;

		public string foo2;

		public STRTH2()
		{
		}

		public STRTH2(STRTH3 a3)
		{
			h3 = a3;
			a3.parent = this;
		}

		public STRTH2(string str)
		{
			foo2 = str;
		}

		public STRTH2(STRTH3 a3, string str)
		{
			h3 = a3;
			a3.parent = this;
			foo2 = str;
		}
	}
}
